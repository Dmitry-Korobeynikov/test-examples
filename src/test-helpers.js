import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import { createStoreForTests } from './store/create';

export const renderShallow = (Component, props = {}, initialState = undefined) => {
  const store = createStoreForTests(initialState);
  const context = { store };
  const childContextTypes = { store: PropTypes.shape({}) };
  const shallowWrapper = shallow(<Component {...props} />, { context, childContextTypes });

  // it's connected component
  if (Component.WrappedComponent) {
    // any side effects?
    // if yes then just export mapStateToProps, mapDispatchToProps and mergeProps separately
    // one more option is to create fakeConnect by using jest.mock(...)
    shallowWrapper.connectedProps = shallowWrapper.instance()
      .selectDerivedProps(store.getState(), {}, store);
  }

  return shallowWrapper;
};

export const callRenderProp = (shallowWrapper, childSelector = null, name = 'children') => {
  return (...args) => {
    const innerWrapper = childSelector
      ? shallowWrapper.find(childSelector)
      : shallowWrapper;

    return innerWrapper.renderProp(name)(...args);
  };
};
