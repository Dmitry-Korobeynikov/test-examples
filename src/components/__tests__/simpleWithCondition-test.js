import { renderShallow } from '../../test-helpers';

import SimpleWithConditionComponent from '../simpleWithCondition';

describe('SimpleWithConditionComponent', () => {
  it('should render content by default', () => {
    const wrapper = renderShallow(SimpleWithConditionComponent);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render content if isLoading is false', () => {
    const wrapper = renderShallow(SimpleWithConditionComponent);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render nothing if isLoading is true', () => {
    const isLoading = true;
    const wrapper = renderShallow(SimpleWithConditionComponent, { isLoading });

    expect(wrapper).toMatchSnapshot();
  });
});
