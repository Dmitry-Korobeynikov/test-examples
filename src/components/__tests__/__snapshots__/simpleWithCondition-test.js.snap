// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleWithConditionComponent should render content by default 1`] = `
<div>
  Loaded
</div>
`;

exports[`SimpleWithConditionComponent should render content if isLoading is false 1`] = `
<div>
  Loaded
</div>
`;

exports[`SimpleWithConditionComponent should render nothing if isLoading is true 1`] = `""`;
